fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/join/JoinCommute.java

```diff
@@ -51,51 +54,44 @@ enum SwapType {
 
     @Override
     public Rule build() {
-        return innerLogicalJoin(any(), any()).then(join -> {
-            if (!check(join)) {
-                return null;
-            }
-            boolean isBottomJoin = isBottomJoin(join);
-            if (swapType == SwapType.BOTTOM_JOIN && !isBottomJoin) {
-                return null;
-            }
-
+        return innerLogicalJoin().when(this::check).then(join -> {
```

Comment:
remove any will lead to all children of join is GroupPlan and isBottomJoin will always return false